### PR TITLE
fix: ensure duplicate components are written to unique files

### PR DESCRIPTION
**Pull request by sscaff1**
Duplicate files are written to unique paths. I mean all I do is append an index to them. I'm not sure if that's best. I tested this and it works fine. Do you have any recommendations on how to make the file unique other than an index? I packed this up locally and tried it in my project where we have a lot of duplicate files and it works fine. Let me know if you need me to do anything else. 

This is the same as #20 which can likely be closed.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -105,11 +105,46 @@ export default function plugin(
...
+              fileName += `${alreadyProcessed.length}`;
+              console.warn(
+                `'${component.filePath}' will be written to '${fileName}.json'`,
+              );
+              return [
+                {
+                  ...processed,
+                  [componentName]: [...alreadyProcessed, component.filePath],
+                },
+                [...components, { fileName, component }],
```
Comment (atomicpages) [inline src/index.ts:133]: No need for a return here since it returns on the next line. Other than this it LGTM

File: src/index.ts [hunk 2]
```diff
@@ -105,11 +105,46 @@ export default function plugin(
...
+              {
+                ...processed,
+                [componentName]: [component.filePath],
+              },
+              [...components, { fileName, component }],
+            ];
+          },
+          [{}, []],
+        );
+        void toProcess[1].map(({ fileName, component }) =>
```
Comment (atomicpages) [inline src/index.ts:146]: Use `.forEach`

**Review by atomicpages — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (sscaff1) [reply src/index.ts]: Updated

File: src/index.ts (hunk 1 above)
Comment (sscaff1) [reply src/index.ts]: Updated

**Comment by sscaff1** [comment]
Updated the code based on comments. I ran the build and pack again and tested it locally with my docusaurus site - works great.

**Review by atomicpages — APPROVED** [review]
Thanks for the contrib! 🙏

**Pull request merged**